Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
+    puts ""
+    puts "Begin Generating Custom GPG Keys"
+    puts "====================================================================================="
+
+    sh "gpg --full-generate-key", verbose: false
+  end
+
+  #Generate new default GPG key
+  task :generate_default do
+    puts ""
+    puts "Begin Generating Default GPG Keys"
+    puts "====================================================================================="
+
+    sh "gpg --generate-key", verbose: false
+  end
+
+  #prompt developer to send GPG key to Platform
+  task :finish do
+    puts "====================================================================================="
+    puts "Key Generation Complete!"
+    puts "Please send your GPG public key to Platform 9-3/4"
+    puts "You can contact them in the Slack channel #platform9"
+    puts "====================================================================================="
+  end
+
+  #ask user if they want to create a key, loop till given a valid answer
+  def create_gpg_key?
+    puts "====================================================================================="
+    puts "To access production credentials for the WordPress app you will need to a GPG Key"
+    puts "Do you need to generate a new GPG Key?"
+    puts "Press 'Y' to create a new key.  Press 'N' to skip"
+
+    display_prompt_response
+  end
+
+  #ask user if they want to create a key,  loop till given a valid answer
+  def create_default_key?
+    puts "====================================================================================="
+    puts "You can choose to setup with a default or custom key pair setup"
+    puts "Default setup - Type: RSA to RSA, RSA length: 2048, Valid for: does not expire"
+    puts "Would you like to continue with the default setup?"
+    puts "====================================================================================="
+    puts "Press 'Y' for Yes.  Press 'N' for custom configuration"
+
+    display_prompt_response
+  end
+
+  #prompt for a Y or N response, continue asking if other character
+  #return true for Y and false for N
+  def display_prompt_response
+    response = STDIN.gets.strip.upcase
+    until response == "Y" || response == "N"
+        puts "Invalid entry, please enter Y or N"
+        response = STDIN.gets.strip.upcase
+    end
+
+    if response == "Y"
+      return true
+    elsif response == "N"
+      return false
+    end
+  end
+
+  #display prompt for developer to aid in setting up default key
+  def display_default_config_helpers
+    puts ""
+    puts ""
+    puts "====================================================================================="
+    puts "You will need to enter the following info to create your key"
+    puts "Please enter your real name, email address, and a password for your key when prompted"
+    puts "====================================================================================="
+  end
+
+#end namespace GPG_key_setup
+end
```

Comment:
While I appreciate the information value these "end namespace" comments give when scanning the code, I do wonder whether the cost of maintaining them or them getting out of date (e.g., the `namespace` here is called `:gpg_key` not `:GPG_key_setup`) is too high.

Reply:
Great point, I hadn't considered that at the time.